src/main/java/us/abstracta/wiresham/VirtualTcpBase.java

```diff
@@ -0,0 +1,118 @@
+package us.abstracta.wiresham;
+
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import javax.net.ssl.SSLContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class VirtualTcpBase<T> {
+
+  public static final int DEFAULT_READ_BUFFER_SIZE = 2048;
+  public static final int DYNAMIC_PORT = 0;
+
+  private static final Logger LOG = LoggerFactory.getLogger(VirtualTcpBase.class);
+
+  protected final Set<ConnectionFlowStepsDriver> connectionDrivers = new HashSet<>();
+  protected final Map<List<PacketStep>, T> packetsByPortMap = new LinkedHashMap<>();
```

Comment:
Sounds weird to have a map from many to one... maybe do it the other way? Is this structure necessary? maybe the flows having the map of port number to socket is enough and they manage that? and at this point you only manage server sockets, and create flows or assign sockets (not server sockets) to existing flows?

Reply:
It says packetsByPort, but it looks more like socketByPackets